core/state_transition.go

```diff
@@ -190,7 +190,12 @@ func AccessListGas(rules params.Rules, accessList types.AccessList) (uint64, err
 	for i, accessTuple := range accessList {
 		address := accessTuple.Address
 		if !rules.PredicateExists(address) {
-			totalGas, overflow := commonMath.SafeAdd(gas, params.TxAccessListAddressGas+uint64(len(accessTuple.StorageKeys))*params.TxAccessListStorageKeyGas)
+			// Previous access list gas calculation does not use safemath because an overflow would not be possible with
+			// the size of access lists that could be included in a block and standard access list gas costs.
+			// Therefore, we only check for overflow when adding to [totalGas], which could include the sum of values
+			// returned by a predicate.
```

Comment:
Could we clarify here or elsewhere what happens if we overflow?

Reply:
I don't think this is necessary. If you look at `IntrinsicGas` it can already return an overflow error, which invalidates the transaction.

I'm going to change the returned error to be the same as the overflow error used in `IntrinsicGas`